src/Database/Driver/Sqlite.php

```diff
@@ -132,12 +134,27 @@ public function connect(): bool
             );
         }
 
-        $databaseExists = file_exists($config['database']);
+        $chmodFile = false;
+        if ($config['database'] !== ':memory:' && $config['mode'] !== 'memory') {
+            $chmodFile = !file_exists($config['database']);
+        }
 
-        $dsn = "sqlite:{$config['database']}";
-        $this->_connect($dsn, $config);
+        $params = [];
+        if ($config['cache']) {
+            $params[] = 'cache=' . $config['cache'];
+        }
+        if ($config['mode']) {
+            $params[] = 'mode=' . $config['mode'];
```

Comment:
What does mode do?

Reply:
https://www.sqlite.org/uri.html Section 3.3

There is a `memory` mode that let's you name the db but act like `:memory:` so you can create multiple in-memory dbs.